### ISSUE TITLE
Conflict in slj4 with the HDF5 viewer?

### ISSUE THREAD

**Issue by MarkRivers** [epics-base/epicsCoreJava#114]
The areaDetector ADViewers/ImageJ plugins are built with the epics-pvaccess and jca .jar files from this module.

We have found that when we try to use those ImageJ plugns with this one: https://github.com/paulscherrerinstitute/ch.psi.imagej.hdf5 
there is a conflict.  

We get this error when launching the HDF5 viewer:
```
ImageJ 1.53e; Java 1.8.0_172 [64-bit]; Linux 4.15.0-123-generic; 87MB of 85867MB (<1%)
java.lang.NoSuchMethodError: org.slf4j.Logger.trace(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/Object;)V
at hdf.object.FileFormat.(FileFormat.java:315)
at hdf.object.h5.H5File.(H5File.java:174)
at hdf.object.h5.H5File.(H5File.java:127)
at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
at java.lang.Class.newInstance(Class.java:442)
at hdf.object.FileFormat.(FileFormat.java:228)
at ch.psi.imagej.hdf5.HDF5Reader.open(HDF5Reader.java:82)
at ch.psi.imagej.hdf5.HDF5Reader.open(HDF5Reader.java:51)
at ch.psi.imagej.hdf5.HDF5Reader.run(HDF5Reader.java:36)
at ij.IJ.runUserPlugIn(IJ.java:240)
at ij.IJ.runPlugIn(IJ.java:203)
at ij.Executer.runCommand(Executer.java:150)
at ij.Executer.run(Executer.java:68)
at java.lang.Thread.run(Thread.java:748)
```

When the EPICS_areaDetector folder is removed, the HDF Viewer goes back to working normally.

This is discussed more in this Github issue: https://github.com/areaDetector/ADViewers/issues/18

Are these EPICS modules using a version of slj4 that is incompatible with the one being used in that HDF5 plugin?

**Comment by kasemir** [comment]
For what it's worth, if any of the EPICS code modules use logging, I suggest they stick with the default java.util.logging package, avoiding any external logging library.

**Comment by shroffk** [comment]
`jca` only depends on hamcrest and junit it has no dependency on slj4.

With `epics-pvaccess` has a **test** dependency to slf4j 1.6.1, since this is a **test** dependency it should not be part of the jar or be required to build or run `epics-pvaccess`. 
How are you packaging and using epics-pvaccess?

**Comment by MarkRivers** [comment]
I figured out the problem, and it is not with the epicsCoreJava files.  The problem is with the NetCDF_.jar file that is included in ADViewers.  That jar file from 2014 uses slf4j, and must use a version that is incompatible with the one used in the HDF5 plugin.  The NetCDF_.jar file is not needed for ADViewers, I provided it as a convenience.  Removing it from ADViewers fixes the issue, and the ADViewers and HDF5 plugin both work.
